mm/ktsan/thr.c

```diff
@@ -38,9 +39,13 @@ kt_thr_t *kt_thr_create(kt_thr_t *thr, int kid)
 		pr_err("KTSAN: maximum number of threads is reached\n");
 		BUG();
 	}
+	/* Kernel does not assign PIDs to some threads, give them fake
+	 * negative PIDs so that they are distinguishable in reports. */
+	if (pid == 0)
+		pid = --pool->new_pid;
```

Comment:
Idle threads have PID 0, but the init thread has PID 0 as well.
I think we want to assign fake PIDs to the idle threads, but not to the init one.
One solution would be to fix up thr->pid in ktsan_init after calling kt_thr_create.